### DISCUSSION TITLE
git help needed

### DISCUSSION THREAD

**Discussion by devesh-mishra13** [freeCodeCamp/Developer_Quiz_Site#788] [Q&A]
I have already made 3 commits which were merged later but still on my local repo which i forked shows 3 commits ahead what can i do to start a new commit. As it shows commits when i will make new branch and push it to origin it will be pushed with the existing commits so what can i do.

**Comment by devesh-mishra13** [comment]
Also if i am syncing fork it shows up to date and the only thing i can do is either first create a pull request of the exisiting commits So please anyone help me out on this issue.

**Comment by jdwilkin4** [comment]
HI @devesh-mishra13 ! 

I have taken a look at your forked copy and there are two ways you can fix this

No.1:
You can read through this github discussion on how to get your main branch properly synced again
https://gist.github.com/seLain/8bdf12c7196f3fccafdf067dec2696b2

No.2:
You can delete your fork
Create a new fork of the Developer Quiz site.
Then you will be back to being up to date and have a clean main branch.

Then each time you are interested in adding new questions and creating a PR, you will first create a new branch.
That will keep your main branch clean so you won't run into these issues.

Hope that helps

**Comment by jdwilkin4** [comment]
So here would be your ideal workflow for being able to do multiple contributions to the same repository

1. update the main branch
2. create a new branch from there
3. make some changes
4. push up those changes
5. create a pr
6. get your pr reviewed, approved and merged
7. switch to the main branch and delete the branch you just created because you don't need it anymore
8. update your main branch
9. create a new branch with new changes and repeat the process all over again

**Reply by devesh-mishra13** [reply]
Thank You so much jdwilkin4 for this workflow it will definitely help me
